Contrib/MLpackages/GlobalAncestry.py

```diff
@@ -0,0 +1,147 @@
+#Global Ancestry with 1k genomes
+
+import pandas as pd
+import numpy as np
+import torch
+import sys
+
+from sklearn import datasets, linear_model
+from sklearn.metrics import mean_squared_error, r2_score
+from sklearn.preprocessing import OneHotEncoder
+from sklearn.linear_model import LinearRegression
+
+#Load the vcf file and ancestry information file
+#genofile = '1kGPfiltered/pruned500k1kGP.vcf'
+#AncstFile = 'igsr_IDwAncst_subset-Copy1.txt'
+#AncstFile = 'igsr_IDwAncst.txt'
+genofile = open(sys.argv[1],'r')
+AncstFile = open(sys.argv[2],'r')
+log = open(sys.argv[3],'w')
+filename = sys.argv[4]
+
+
+
+#Read in both files
+headervcf=pd.read_csv(genofile,delimiter='\t',skiprows=25)
+IDswAncst = pd.read_csv(AncstFile,delimiter = ',')
+
+
+
+#Turn genotypes into dosage format for PCA
+dosagevcf = headervcf.replace('0/0', 0)
+dosagevcf = dosagevcf.replace('0/1', 1)
+dosagevcf = dosagevcf.replace('1/0', 1)
+dosagevcf = dosagevcf.replace('1/1', 2)
+
+
+#Separate the genotype dosages
+IDs = headervcf.iloc[1]
+ChromPos = dosagevcf.iloc[:, 0:9]
+Genos = dosagevcf.iloc[:, 9:]
+
+
+#Separate IDs out as a column by just pulling out one row
+IDsascol = Genos.iloc[1]
+#Transpose the headers into a column
+IDsascol = IDsascol.T.reset_index()
+#Pull out just the converted header column
+IDsascol = IDsascol.iloc[:,0]
+
+
+
+#Convert ints to floats and remove excess whitespace for converting to tensor
+Genos = Genos.astype(float)
+Genos.columns = Genos.columns.str.strip()
+
+
+
+#Find the rows with missing data
+missing_mask = np.isnan(Genos).any(axis=1)
```

Comment:
how much data are we missing here?   IIRC 1KGP is pretty clean [?] We should definitely at least be logging the variants we're discarding due to missingness.